android/app/src/main/kotlin/com/elodin/walkie_talkie/GattClientManager.kt

```
@@ -199,18 +295,26 @@ class GattClientManager(
+            Log.i(TAG, "Connecting to host at $macAddress (attempt ${connectRetryCount + 1})")
+            pendingMacAddress = macAddress
             gatt = device.connectGatt(context, false, gattCallback, BluetoothDevice.TRANSPORT_LE)
             return gatt != null
-        } catch (e: SecurityException) {
-            Log.e(TAG, "Missing Bluetooth permissions for connectGatt", e)
-            onError?.invoke("BLUETOOTH_PERMISSION_DENIED")
-            return false
-        } catch (e: IllegalArgumentException) {
-            Log.e(TAG, "Invalid MAC address: $macAddress", e)
-            return false
         } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)